src/file_compound.h

```diff
@@ -87,20 +87,8 @@ class LIBZIM_PRIVATE_API FileCompound : private std::map<Range, FilePart*, less_
     }
 
     PartRange locate(offset_t offset, zsize_t size) const {
-#if ! defined(__APPLE__)
-        return equal_range(Range(offset, offset+size));
-#else
-        // Workaround for https://github.com/openzim/libzim/issues/398
-        // Under MacOS the implementation of std::map::equal_range() makes
-        // assumptions about the properties of the key comparison function and
-        // abuses the std::map requirement that it must contain unique keys. As
-        // a result, when a map m is queried with an element k that is
-        // equivalent to more than one keys present in m,
-        // m.equal_range(k).first may be different from m.lower_bound(k) (the
-        // latter one returning the correct result).
-        const Range queryRange(offset, offset+size);
-        return {lower_bound(queryRange), upper_bound(queryRange)};
-#endif // ! defined(__APPLE__)
+      const Range queryRange(offset, offset+size);
+      return {lower_bound(queryRange), upper_bound(queryRange)};
```

Comment:
1. Regarding the commit message - it refers to the [`std::equal_range()`](https://en.cppreference.com/w/cpp/algorithm/equal_range) algorithm whereas here [`std::map<K,T,C,A>::equal_range()`](https://en.cppreference.com/w/cpp/container/map/equal_range) member function is used which doesn't have the said restriction in the spec.

2. It would be nice to preserve an adjusted comment explaining why `lower_bound() + upper_bound()` is used instead of `equal_range()`.

Reply:
> Regarding the commit message - it refers to the [std::equal_range()](https://en.cppreference.com/w/cpp/algorithm/equal_range) algorithm whereas here [std::map<K,T,C,A>::equal_range()](https://en.cppreference.com/w/cpp/container/map/equal_range) member function is used which doesn't have the said restriction in the spec.

Indeed. But from https://stackoverflow.com/questions/67042750/should-setequal-range-return-pair-setlower-bound-setupper-bound it seems we have the same issue.

And this is confirmed with https://gist.github.com/mgautierfr/dd2cb595cec4bb9c569244187c88c00d

> It would be nice to preserve an adjusted comment explaining why lower_bound() + upper_bound() is used instead of equal_range().

Added